tests/python/test_benchmark_elements.py

```diff
@@ -0,0 +1,466 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022-2025 The ImpactX Community
+#
+# Authors: Axel Huebl
+# License: BSD-3-Clause-LBNL
+#
+# This set of tests is used for performance benchmarking of individual
+# elements pushes (as micro-benchmarks). We use this file to rapidly
+# evaluate performance changes when tuning beamline element performance
+# on CPUs and GPUs.
+#
+# -*- coding: utf-8 -*-
+
+from functools import partial
+
+import pytest
+
+from impactx import ImpactX, Map6x6, distribution, elements
+
+# benchmark config
+rounds = 5
+npart = 1_000_000  # increase this to >10M or even 100M to avoid L1/L2/L3 cache effects on some hardware.
+
+# element config
+nslice = 1
+mapsteps = 4
+
+
+@pytest.fixture(scope="function")
+def sim():
+    class SimContextManager:
+        def __enter__(self):
+            self.sim = ImpactX()
+
+            # set numerical parameters and IO control
+            self.sim.particle_shape = 2  # B-spline order
+            self.sim.space_charge = False
+            self.sim.diagnostics = False  # benchmarking
+            self.sim.slice_step_diagnostics = False
+
+            self.sim.init_grids()
+
+            beam = self.sim.particle_container()
+            beam.clear_particles()
+
+            # load a 2 GeV electron beam with an initial
+            # unnormalized rms emittance of 2 nm
+            kin_energy_MeV = 2.0e3  # reference energy
+            bunch_charge_C = 1.0e-9  # used with space charge
+
+            #   reference particle
+            ref = self.sim.particle_container().ref_particle()
+            ref.set_charge_qe(-1.0).set_mass_MeV(0.510998950).set_kin_energy_MeV(
+                kin_energy_MeV
+            )
+
+            #   particle bunch
+            distr = distribution.Waterbag(
+                lambdaX=3.9984884770e-5,
+                lambdaY=3.9984884770e-5,
+                lambdaT=1.0e-3,
+                lambdaPx=2.6623538760e-5,
+                lambdaPy=2.6623538760e-5,
+                lambdaPt=2.0e-3,
+                muxpx=-0.846574929020762,
+                muypy=0.846574929020762,
+                mutpt=0.0,
+            )
```

Comment:
We could make this simpler:
Maybe we can use simple numbers for twiss parameters of the distribution and a 1 GeV beam or so.